controller/update_test.go

```diff
@@ -228,40 +210,65 @@ func TestIsGroupRemovalAllowed(t *testing.T) {
 // TestUpdateWithStrategy tests the UpdateWithStrategy method.
 func TestUpdateWithStrategy(t *testing.T) {
 	tests := []struct {
-		fleetMockSetUp func(*fleetMock)
-		ctx            context.Context
-		req            Request
-		opts           UpdateOptions
-		errMatcher     func(error) bool
+		fleetSetUp func(fleet.Fleet)
+		req        Request
+		opts       UpdateOptions
+		assertion  func(*testing.T, *fleet.DummyFleet, error)
 	}{
-		// Test that IsWaitTimeoutReached is returned if we attempt to
-		// update given nil arguments.
+		// Test a minimal update.
 		{
-			fleetMockSetUp: nil,
-			ctx:            context.Background(),
-			req:            Request{},
-			opts:           UpdateOptions{},
-			errMatcher:     IsWaitTimeoutReached,
+			fleetSetUp: func(f fleet.Fleet) {
+				unitName := "bluebird-unit@1.service"
+
+				f.Submit(context.Background(), unitName, "some content")
+				f.Start(context.Background(), unitName)
+			},
+			req: Request{
+				RequestConfig: RequestConfig{
+					Group:    "bluebird",
+					SliceIDs: []string{"1"},
+				},
+				Units: []Unit{
+					Unit{
+						Name:    "bluebird-unit@.service",
+						Content: "some updated content",
+					},
+				},
+			},
+			opts: UpdateOptions{
+				MaxGrowth: 1,
+				MinAlive:  0,
+			},
+			assertion: func(t *testing.T, f *fleet.DummyFleet, e error) {
+				if e != nil {
+					t.Fatal("Error returned by update:", e)
+				}
+
+				unitStatusList, err := f.GetStatusWithMatcher(
+					func(s string) bool {
+						return strings.HasPrefix(s, "bluebird-unit@") && strings.HasSuffix(s, ".service")
+					},
+				)
+				if err != nil {
+					t.Fatal("Error returned getting statuses: ", err)
+				}
+				if len(unitStatusList) != 1 {
+					t.Fatal("Incorrect number of units:", len(f.Units))
+				}
+				if unitStatusList[0].Name == "bluebird-unit@1.service" {
```

Comment:
a) I'm not a _massive_ fan of gomega, to be honest
b) we should do that, i'll add it